data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -177,6 +190,23 @@ private boolean isAddressDefinedLocally(final String address) {
         }
     }
 
+    private List<Record<Event>> batchRecordsForForwarding(final String destinationIp, final List<Record<Event>> records) {
+        if (!peerBatchingQueueMap.containsKey(destinationIp)) {
+            peerBatchingQueueMap.put(destinationIp, new LinkedBlockingQueue<>(forwardingBatchSize * 5));
+        } else {
+            peerBatchingQueueMap.get(destinationIp).addAll(records);
+        }
+
+        if (peerBatchingQueueMap.get(destinationIp).size() >= forwardingBatchSize) {
```

Comment:
We should have a configurable timeout. Otherwise with low volumes of data these batch could get stuck.

The old peer-forward had one.

https://github.com/opensearch-project/data-prepper/tree/1.5.2/data-prepper-plugins/peer-forwarder#configuration

Let's use the Data Prepper duration for it.

```
forwarding_batch_timeout: 3s
```

Reply:
Good point. I will add that